samples/csharp_dotnetcore/13.core-bot/Dialogs/RootDialog.cs

```diff
@@ -0,0 +1,241 @@
+                        {
+                            new SendActivity("Sure, cancelling that..."),
+                            new CancelAllDialogs(),
+                            new EndDialog()
+                        }
+                    },
+                    new OnIntent()
+                    {
+                        Intent = "Help",
+                        Condition = "#Help.Score >= 0.8",
+                        Actions = new List<Dialog>() {new SendActivity("${DisplayHelp()}")}
+                    },
+                    new OnIntent()
+                    {
+                        Intent = "Greeting",
+                        Actions = new List<Dialog>() {new SendActivity("${BotOverview()}")}
+                    },
+                    new OnUnknownIntent()
+                    {
+                        Actions = new List<Dialog>() {new SendActivity("${UnknownIntent()}")}
+                    },
+                    new OnIntent("Book_flight")
+                    {
+                        Actions = BookFlightSteps()
+                    },
+                    new OnEndOfActions()
+                    {
+                        Actions = BookFlightSteps()
+                    }
+                }
+            };
+
+            // Add named dialogs to the DialogSet. These names are saved in the dialog state.
+            AddDialog(rootDialog);
+
+            // The initial child Dialog to run.
+            InitialDialogId = nameof(AdaptiveDialog);
+        }
+
+        private static List<Dialog> WelcomeUserSteps()
+        {
+            return new List<Dialog>()
+            {
+                // Iterate through membersAdded list and greet user added to the conversation.
+                new Foreach()
+                {
+                    ItemsProperty = "turn.activity.membersAdded",
+                    Actions = new List<Dialog>()
+                    {
+                        // Note: Some channels send two conversation update events - one for the Bot added to the conversation and another for user.
+                        // Filter cases where the bot itself is the recipient of the message. 
+                        new IfCondition()
+                        {
+                            Condition = "$foreach.value.name != turn.activity.recipient.name",
+                            Actions = new List<Dialog>()
+                            {
+                                new SendActivity("${WelcomeCard()}")
+                            }
+                        }
+                    }
+                },
+            };
+        }
+
+        private static Recognizer CreateRecognizer(IConfiguration configuration)
+        {
+            if (string.IsNullOrEmpty(configuration["LuisAppId"]) || string.IsNullOrEmpty(configuration["LuisAPIKey"]) ||
+                string.IsNullOrEmpty(configuration["LuisAPIHostName"]))
+            {
+                return new RegexRecognizer
```

Comment:
RegEx if LUIS is not configured seems like a great idea.

We should maybe also have a message stating how to setup luis, similar to the waterfall version of corebot:
![image](https://user-images.githubusercontent.com/11055362/95785378-7ef5f300-0c8a-11eb-8598-94a64eaf8615.png)